src/UI/Components/TreeTable/TreeRow/TreeRowView.tsx

```diff
@@ -1,74 +1,116 @@
-import React from "react";
-import { Icon, Split, SplitItem, Tooltip } from "@patternfly/react-core";
-import { ExclamationTriangleIcon } from "@patternfly/react-icons";
+import React, { FC } from "react";
+import {
+  Button,
+  Icon,
+  Split,
+  SplitItem,
+  Tooltip,
+} from "@patternfly/react-core";
+import { ExclamationTriangleIcon, InfoAltIcon } from "@patternfly/react-icons";
 import { Tr, Td } from "@patternfly/react-table";
 import styled from "styled-components";
-import { ParsedNumber, Attributes } from "@/Core";
+import { ParsedNumber, Attributes, AttributeAnnotations } from "@/Core";
 import { Toggle } from "@/UI/Components/Toggle";
 import { ClipboardCopyButton } from "../../ClipboardCopyButton";
 import { CellWithCopy } from "./CellWithCopy";
 import { CellWithCopyExpert } from "./CellWithCopyExpert";
 import { Indent } from "./Indent";
 import { TreeRow } from "./TreeRow";
 
+/**
+ * RowProps is an interface that defines the props of the TreeRowView component.
+ */
 interface RowProps {
   row: TreeRow;
   id: string;
   serviceEntity: string;
   version: ParsedNumber;
   showExpertMode: boolean;
   attributes: Attributes;
+  annotations?: AttributeAnnotations;
+  setTab?: (tab: string) => void;
 }
 
 const warningMessage =
   "This attribute migrated to a different/new Type and can’t be displayed properly into the table. You can copy the object for further comparison through the Copy button. It will store the value of each state in your clipboard.";
 
+/**
+ * TreeRowView is a component that renders a row in the tree table for the Attributes.
+ *
+ * @param {RowProps} props - The props of the component.
+ *  @prop {TreeRow} row - The row object.
+ *  @prop {string} id - The id of the row.
+ *  @prop {string} serviceEntity - The service entity.
+ *  @prop {ParsedNumber} version - The version number.
+ *  @prop {boolean} showExpertMode - The flag to show the expert mode.
+ *  @prop {Attributes} attributes - The attributes object.
+ *  @prop {AttributeAnnotations} annotations - The annotations object, optional.
+ *  @prop {function} setTab - The callback for setting the active tab, optional.
+ *
+ * @returns The tree row view component.
+ */
 export const TreeRowView: React.FC<RowProps> = ({
   row,
   id,
   serviceEntity,
   version,
   showExpertMode,
   attributes,
+  annotations,
+  setTab = () => {},
 }) => {
   switch (row.kind) {
     case "Flat":
       return (
         <StyledTr aria-label={`Row-${row.id}`}>
           <Td dataLabel={row.primaryCell.label}>
             <Indent level={0} noToggle>
-              {row.primaryCell.value}
+              {annotations?.web_presentation !== "documentation" ? (
+                row.primaryCell.value
+              ) : (
+                <DocumentationCell
```

Comment:
I think that the gain for this kind of components is minuscule. it doesn't save that much lines of code there, there is nothing extra happening inside of the component itself, and it's not reused anywhere else.

That doesn't mean that I want to remove it, just sharing my opinion

Reply:
I separated it because when I see ternaries with big blobs of code, I lose sight of what I'm reading. Naming the component also adds readability.